### ISSUE TITLE
FireQOS install / tutorial out of date?

### ISSUE THREAD

**Issue by twkonefal** [firehol/firehol#194]
I'm trying to follow the [Tutorial](https://github.com/firehol/firehol/wiki/FireQOS-Tutorial), but I've run into some difficulties.

```
PREPARATION: Download or Update FireQOS
Go to the directory you want to download FireQOS and run:
git clone https://github.com/ktsaou/firehol.git firehol.git
cd firehol.git
git checkout origin/master
If you already have done this, you can update it any time with this:

cd firehol.git
git checkout origin/master
git pull origin master
Now you have fireqos.in and firehol.in the in the sbin directory.
```

Except that I don't actually have it.

```
# ls -alF ./sbin/
total 816
drwxr-xr-x.  2 root root   4096 Jan 24 14:45 ./
drwxr-xr-x. 16 root root   4096 Jan 24 14:45 ../
-rwxr-xr-x.  1 root root 364335 Jan 24 14:45 firehol*
-rwxr-xr-x.  1 root root 104191 Jan 24 14:45 fireqos*
-rw-r--r--.  1 root root   2095 Jan 24 14:45 functions.common
-rw-r--r--.  1 root root   2026 Jan 24 14:45 install.config.in.in
-rwxr-xr-x.  1 root root  47660 Jan 24 14:45 link-balancer*
-rw-r--r--.  1 root root    981 Jan 24 14:45 Makefile.am
-rwxr-xr-x.  1 root root 270908 Jan 24 14:45 update-ipsets*
-rwxr-xr-x.  1 root root  20095 Jan 24 14:45 vnetbuild*
```
Also, nothing actually works. Running ./sbin/fireqos does this:

```# ./sbin/fireqos
Cannot access ./sbin/install.config
```

So... how do I get this working?

**Comment by philwhineray** [comment]
Yes, it looks like it needs a refresh: thanks for pointing it out.

To get the programme working, run ./configure and make. Check out the
readme in the repo you cloned.

**Comment by philwhineray** [comment]
Actually, you might need to do more than that if you clone as suggested in
the tutorial. It might be easier to get a package from the github release
page; those definitely just need configure and make.


On 24 Jan 2017 10:58 p.m., "Phil Whineray" <phil.whineray@gmail.com> wrote:

Yes, it looks like it needs a refresh: thanks for pointing it out.

To get the programme working, run ./configure and make. Check out the
readme in the repo you cloned.

**Comment by ktsaou** [comment]
I suggest to use this to install all of firehol: https://github.com/firehol/netdata-demo-site/blob/master/install-all-firehol.sh

It downloads sources into `/usr/src/` and installs all tools (firehol, fireqos, iprange, netdata) system-wide.

**Comment by ktsaou** [comment]
updated the wiki page too.
@philwhineray could you please take care of the site?

**Comment by twkonefal** [comment]
Thank you!

Incidentally, is there any way to get FireQOS installed without building from source? (./configure; make; etc). The reason I ask is because I'm running a minimal server and would prefer very much to not have to install a development environment/compiler onto it. (Sorry if this is off-topic)

**Comment by ktsaou** [comment]
I am sorry. No currently. FireQOS is a shell script and could be done, but we have spent quite some time to adapt to autotools for providing compatibility with what the distros expect.

@philwhineray is there a way to run it without `./configure`

**Comment by philwhineray** [comment]
You need the `install.config` in the same folder as the `fireqos` script. You could generate this file on a separate system and then copy them together, along with `functions.common`. The entries in `/sbin` can be softlinks, so you can give them their own folder if you want.

@twkonefal you don't say what packaging format. Debian provide have an up-to-date package in testing. It would appear that RHEL maintainer has gone so there is no longer an RPM in the official repos so we build one [here](https://github.com/firehol/packages/releases).

Or try [checkinstall](https://help.ubuntu.com/community/CheckInstall) to build your own package.

**Comment by kcsapersona** [comment]
Ran into a similar issue when attempting to run the update-ipsets command.
Cannot access ./install.config.

The script update-ipsets says: 
# This script depends on iprange, found also in firehol.
# It does not depend on firehol. You can use it without firehol.

But installing iprange by itself does not include the update-ipsets script.
If I try copy the update-ipsets script to my iprange server (installed without firehol),
and try to run it, it says: "ERROR : Cannot find a working iprange command. It should be part of FireHOL but it is not installed."

So it appears that the update-ipsets script is "Dependent" on firehol. Or am I missing something?
Sorry if this is off topic, but it seemed similar to the thread above.

**Comment by ktsaou** [comment]
To use update-ipsets, you need to have firehol and iprange installed. But you don't need to run a firehol managed firewall.

firehol, fireqos, update-ipsets and the rest of the tools at the firehol package, are just shell scripts (meaning that they are not some kind of heavy applications that occupy significant disk space).

**Comment by kcsapersona** [comment]
Thank you for the clarification.
We pretty much thought this was the case.
Very Best
